addons/website_livechat/static/src/models/visitor/visitor.js

```diff
@@ -0,0 +1,67 @@
+odoo.define('website_livechat/static/src/models/partner/partner.js', function (require) {
+'use strict';
+
+const { registerNewModel } = require('mail/static/src/model/model_core.js');
+const { attr, many2one, one2many } = require('mail/static/src/model/model_field.js');
+
+function factory(dependencies) {
+
+    class Visitor extends dependencies['mail.model'] {
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('country_id' in data && data.country_id) {
+                data2.country = [['insert', {
+                    id: data.country_id,
+                    code: data.country_code,
+                }]];
+            }
+            if ('history' in data) {
+                data2.history = data.history;
+            }
+            if ('is_connected' in data) {
+                data2.is_connected = data.is_connected;
+            }
+            if ('lang' in data) {
+                data2.lang = data.lang;
+            }
+            if ('name' in data) {
+                data2.name = data.name;
+            }
+            if ('partner_id' in data && data.partner_id) {
+                data2.partner = [['insert', { id: data.partner_id }]];
+            }
+            if ('website' in data) {
+                data2.website = data.website;
+            }
+            return data2;
+        }
+    }
+
+    Visitor.fields = {
+        country: many2one('mail.country'),
+        history: attr(),
+        is_connected: attr(),
+        lang: attr(),
+        name: attr(),
+        partner: many2one('mail.partner'),
+        threads: one2many('mail.thread', {
+            inverse: 'visitor',
+        }),
+        website: attr(),
```

Comment:
Compute most of those fields as defaults to the related partner value when the partner is set?